envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/MetadataNodeGroup.kt

```diff
@@ -15,16 +19,121 @@ class MetadataNodeGroup(val properties: SnapshotProperties) : NodeGroup<Group> {
         return createGroup(node, ads)
     }
 
+    @SuppressWarnings("ReturnCount")
+    private fun metadataToListenersHostPort(
+        id: String,
+        ingressHostValue: Value?,
+        ingressPortValue: Value?,
+        egressHostValue: Value?,
+        egressPortValue: Value?
+    ): ListenersHostPortConfig? {
+        if (listOf(ingressHostValue, ingressPortValue, egressHostValue, egressPortValue).all { it == null }) {
+            logger.debug("Node $id with static listener config connected.")
+            return null
+        }
+
+        if (ingressHostValue == null) {
+            logger.info("Node $id has no ingress host configured, falling back to static listeners.")
+            return null
+        }
+
+        if (ingressPortValue == null) {
+            logger.info("Node $id has no ingress port configured, falling back to static listeners.")
+            return null
+        }
+
+        if (egressHostValue == null) {
+            logger.info("Node $id has no egerss host configured, falling back to static listeners.")
+            return null
+        }
+
+        if (egressPortValue == null) {
+            logger.info("Node $id has no egress port configured, falling back to static listeners.")
```

Comment:
Shouldn't these 4 above logs be on WARN level? If I see correctly that situation can happen only if we have some bug.

Reply:
I'll change to warn :)